### PR TITLE
Add test for user type interface

### PR DESCRIPTION
**Pull request by yalzhang**
User type interface can have custom ip address set in interface. Add
test cases about setting ip address into interface section.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virtual_network/iface_options.py [hunk 1]
```diff
@@ -423,6 +425,11 @@ def check_mcast_network(session):
     test_guest_ip = "yes" == params.get("test_guest_ip", "no")
     test_backend = "yes" == params.get("test_backend", "no")
     check_guest_trans = "yes" == params.get("check_guest_trans", "no")
+    set_ip = "yes" == params.get("set_user_ip", "no")
+    set_ips = params.get("set_ips", "").split()
+    expect_ip = params.get("expect_ip", None)
+    expect_gw = params.get("expect_gw", None)
+    expect_ns = params.get("expect_ns", None)
```
Comment (chloerh) [inline libvirt/tests/src/virtual_network/iface_options.py]: This is unnecessary. 
expect_ns = params.get("expect_ns") is enough.

File: libvirt/tests/src/virtual_network/iface_options.py [hunk 2]
```diff
@@ -310,24 +313,23 @@ def check_user_network(session):
...
         if len(vm_ips) != len(set(vm_ips)):
-            test.fail("Duplicated IP address on guest. "
-                      "Check bug: https://bugzilla.redhat."
-                      "com/show_bug.cgi?id=1147238")
-
+            logging.debug("Duplicated IP address on guest. Check bug: "
+                          "https://bugzilla.redhat.com/show_bug.cgi?id=1147238")
         for vm_ip in vm_ips:
-            if vm_ip is None or not vm_ip.startswith("10.0.2."):
+            if vm_ip is None or vm_ip != expect_ip:
```
Comment (chloerh) [inline libvirt/tests/src/virtual_network/iface_options.py]: if not vm_ip or vm_ip != expect_ip:

File: libvirt/tests/src/virtual_network/iface_options.py (hunk 1 above)
Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_options.py]: Thank you! have updated the expect_* both to without 'None'

File: libvirt/tests/src/virtual_network/iface_options.py (hunk 2 above)
Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_options.py]: updated

**Comment by chloerh** [comment]
Looks good to me

File: libvirt/tests/src/virtual_network/iface_options.py [hunk 3]
```diff
@@ -310,24 +313,23 @@ def check_user_network(session):
...
-            test.fail("Duplicated IP address on guest. "
-                      "Check bug: https://bugzilla.redhat."
-                      "com/show_bug.cgi?id=1147238")
-
+            logging.debug("Duplicated IP address on guest. Check bug: "
+                          "https://bugzilla.redhat.com/show_bug.cgi?id=1147238")
         for vm_ip in vm_ips:
-            if vm_ip is None or not vm_ip.startswith("10.0.2."):
+            if not vm_ip or vm_ip != expect_ip:
+                logging.debug("vm_ip is %s, expect_ip is %s" % (vm_ip, expect_ip))
```
Comment (dzhengfy) [inline libvirt/tests/src/virtual_network/iface_options.py]: logging.debug("vm_ip is %s, expect_ip is %s", vm_ip, expect_ip)

Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_options.py]: Updated, Thank you!

**Comment by yalzhang** [comment]
depend on https://github.com/avocado-framework/avocado-vt/pull/2290
depend on https://github.com/avocado-framework/avocado-vt/pull/1877

**Pull request merged**